Sources/Protocols/Designable/PaddingDesignable.swift

```diff
@@ -6,53 +6,49 @@
 import UIKit
 
 public protocol PaddingDesignable: class {
+
   /**
-    `padding-left`
-  */
+   `padding-left`
+   */
   var paddingLeft: CGFloat { get set }
 
   /**
-    `padding-right`
-  */
+   `padding-right`
+   */
   var paddingRight: CGFloat { get set }
 
   /**
    `padding-left` and `padding-right`
-  */
+   */
   var paddingSide: CGFloat { get set }
+
 }
 
 public extension PaddingDesignable where Self: UITextField {
-  public func configurePaddingLeft() {
-    if paddingLeft.isNaN {
-      return
-    }
-
-    let padding = UIView(frame: CGRect(x: 0, y: 0, width: paddingLeft, height: 1))
-    leftViewMode = .always
-    leftView = padding
-  }
 
-  public func configurePaddingRight() {
-    if paddingRight.isNaN {
-      return
+  public func paddedRect(forBounds bounds: CGRect) -> CGRect {
+    if paddingSide.isNaN && paddingLeft.isNaN && paddingRight.isNaN {
+      return bounds
     }
-
-    let padding = UIView(frame: CGRect(x: 0, y: 0, width: paddingRight, height: 1))
-    rightViewMode = .always
-    rightView = padding
+    return UIEdgeInsetsInsetRect(bounds, paddingInsets)
   }
 
-  public func configurePaddingSide() {
+  private var paddingInsets: UIEdgeInsets {
     if paddingSide.isNaN {
-      return
+      return insets(left: paddingLeft, right: paddingRight)
```

Comment:
I did the check inside of `insets(left:right:)`. If either `isNaN` it will default to 0. 
I could extract the logic if it's not clear.